nexus/src/app/background/tasks/support_bundle_collector.rs

```diff
@@ -568,153 +568,173 @@ impl BundleCollection<'_> {
         {
             report.listed_in_service_sleds = true;
 
-            // NOTE: This could be, and probably should be, done concurrently.
-            for sled in &all_sleds {
-                info!(&log, "Collecting bundle info from sled"; "sled" => %sled.id());
-                let sled_path = dir
-                    .path()
-                    .join("rack")
-                    .join(sled.rack_id.to_string())
-                    .join("sled")
-                    .join(sled.id().to_string());
-                tokio::fs::create_dir_all(&sled_path).await?;
-                tokio::fs::write(
-                    sled_path.join("sled.txt"),
-                    format!("{sled:?}"),
-                )
-                .await?;
+            const MAX_CONCURRENT_SLED_REQUESTS: usize = 16;
+            let mut sled_stream = futures::stream::iter(all_sleds)
+                .map(|sled| async move {
+                    self.collect_data_from_sled(&log, &sled, dir).await
+                })
```

Comment:
hm, I note that while the `collect_data_from_sled` calls will execute _concurrently_, the async fns will not actually run in _parallel_, since they're not spawned in separate tasks. so there will only ever be one thread executing one of those futures at a time, although we will be running them  concurrently and executing the next one when one yields. in this case, i think that's mostly okay, since it looks like _most_ of the runtime of `collect_data_from_sled` is probably waiting on IO (waiting for the HTTP requests to come back and the `tokio::fs` operations, which run on the blocking threadpool, to complete). but, i might nonetheless consider spawning tasks for each sled, instead. up to you.

Reply:
(if you decide these should run in _parallel_ i might consider using `tokio::task::JoinSet` rather than `buffer_unordered`, since it uses the task allocation to store the task's membership in the joinset and therefore uses much less heap. may not actually matter though.)

Reply:
That's super interesting, thanks for sharing that bit of info on `buffer_unordered` vs `JoinSet`.

Reply:
Made these changes in 59e8750d8d79d9deecc21046b5869fff9118dc1c - I'm using some manual counting to tasks to respect my previous "maximum bound on current activity".